File: GFG/LinkedList/nthnode.py
import ds

def nthend(dl,n):
    if(dl.head is None or n>dl.size()):
        return
    curr=dl.head
    ncurr=dl.head
    i=0
    while(curr!=None):
        if(i>=n):
            curr=curr.next
            ncurr=ncurr.next
        else:
            curr=curr.next
        i+=1
    return ncurr.data

dll=ds.LinkedList()
for i in range(1,11):
    dll.insert(i*10)
dll.display()
print(nthend(dll,3))
    

File: GFG/LinkedList/nthnode.cpp
#include<iostream>
#include"../ds.h"
using namespace std;

int nthend(LinkedList<int> &dl,int n){
    if(dl.head==NULL || n>dl.size()){
        return 0;
    }
    Node<int>* curr=dl.head;
    Node<int>* ncurr=dl.head;
    int i=0;
    while(curr!=NULL){
        if(i>=n){
            curr=curr->next;
            ncurr=ncurr->next;
        }
        else{
            curr=curr->next;
        }
        i++;
    }
    return ncurr->data;
}

int main(){
    LinkedList<int> dll;
    for (int i=1;i<11;i++){
        dll.insert(i*10);
    }
    dll.display();
    cout<<nthend(dll,3);
    return 0;
}